packages/ceramic-http-client/src/ceramic-http-client.ts

```diff
@@ -65,21 +59,47 @@ class CeramicClient {
     }
   }
 
-  async createDocument (genesis: any, doctype: string, opts?: InitOpts): Promise<Document> {
-    const doc = await Document.create(genesis, doctype, this._apiUrl, opts)
-    if (!this._docmap[doc.id]) this._docmap[doc.id] = doc
-    return doc
+  async createDocument<T extends Doctype>(doctype: string, params: object, opts?: InitOpts): Promise<T> {
+    const doc = await Document.create(this._apiUrl, doctype, params, opts)
+    // if (!this._docmap[doc.id]) {
+      this._docmap[doc.id] = doc
+    // }
+    return doc.doctype as T
+  }
+
+  async loadDocument<T extends Doctype>(id: string, opts?: InitOpts): Promise<T> {
+    if (!this._docmap[id]) {
+      this._docmap[id] = await Document.load(id, this._apiUrl)
+    }
+    return this._docmap[id].doctype as T
+  }
+
+  get ipfs(): any {
+    throw new Error('method not implemented')
+  }
+
+  addDoctype<T extends Doctype>(doctypeHandler: DoctypeHandler<T>): void {
+    throw new Error('method not implemented')
+  }
```

Comment:
Is this out of scope for now?

Reply:
It's not being used from the `ceramic-http-client` for now. I also renamed it to `addDoctypeHandler` :)